vars/abapEnvironmentPipelineStageBuild.groovy

```diff
@@ -0,0 +1,36 @@
+import groovy.transform.Field
+import com.sap.piper.Utils
+
+import static com.sap.piper.Prerequisites.checkScript
+
+@Field String STEP_NAME = getClass().getName()
+@Field Set GENERAL_CONFIG_KEYS = []
+@Field STAGE_STEP_KEYS = [
+    'cloudFoundryCreateServiceKey',
+    'abapAddonAssemblyKitReserveNextPackages',
+    'abapEnvironmentAssemblePackages',
+    'abapAddonAssemblyKitRegisterPackages',
+    'abapAddonAssemblyKitReleasePackages',
+    'abapAddonAssemblyKitCreateTargetVector',
+    'abapAddonAssemblyKitPublishTargetVector'
+]
+@Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS.plus(STAGE_STEP_KEYS)
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+/**
+ * This stage builds an AddOn for the SAP Cloud Platform ABAP Environment
+ */
+void call(Map parameters = [:]) {
+    def script = checkScript(this, parameters) ?: this
+    def stageName = parameters.stageName?:env.STAGE_NAME
+
+    piperStageWrapper (script: script, stageName: stageName, stashContent: [], stageLocking: false) {
+        cloudFoundryCreateServiceKey script: parameters.script
```

Comment:
Is this step used again in order to create the appropriate user/communication arrangement?

Reply:
yes, the build step needs SAP_COM_0582. In Prepare System, SAP_COM_0510 is created in order to clone/pull the repositories